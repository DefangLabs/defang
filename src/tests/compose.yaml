name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
