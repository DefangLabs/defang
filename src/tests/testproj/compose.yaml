name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      target: testproj
      args:
        DNS: dfnx
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
