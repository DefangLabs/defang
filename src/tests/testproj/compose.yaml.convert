[
  {
    "name": "dfnx",
    "platform": 2,
    "internal": true,
    "deploy": {
      "replicas": 1,
      "resources": {
        "reservations": {
          "memory": 256,
          "cpus": 0.25
        }
      }
    },
    "ports": [
      {
        "target": 80,
        "mode": 1
      },
      {
        "target": 1234,
        "mode": 1
      }
    ],
    "environment": {
      "DOTENV": "enabled",
      "DOT_ENV_INTERPOLATION": "enabled",
      "FOO": "bar"
    },
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "args": {
        "DNS": "dfnx"
      },
      "target": "testproj"
    },
    "secrets": [
      {
        "source": "dummy"
      }
    ],
    "healthcheck": {
      "test": [
        "CMD",
        "curl",
        "-f",
        "http://localhost/"
      ]
    },
    "networks": 1
  }
]