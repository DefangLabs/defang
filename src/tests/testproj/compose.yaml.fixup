[
  {
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "args": {
        "DNS": "dfnx"
      },
      "target": "testproj"
    },
    "command": null,
    "deploy": {
      "replicas": 1,
      "resources": {
        "limits": {
          "cpus": 0.5,
          "memory": "536870912"
        },
        "reservations": {
          "cpus": 0.25,
          "memory": "268435456"
        }
      },
      "placement": {}
    },
    "entrypoint": null,
    "environment": {
      "DOTENV": "enabled",
      "DOT_ENV_INTERPOLATION": "enabled",
      "FOO": "bar",
      "dummy": null
    },
    "healthcheck": {
      "test": [
        "CMD",
        "curl",
        "-f",
        "http://localhost/"
      ]
    },
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 80,
        "protocol": "tcp",
        "app_protocol": "http"
      },
      {
        "mode": "ingress",
        "target": 1234,
        "protocol": "tcp",
        "app_protocol": "http"
      },
      {
        "mode": "host",
        "target": 4567,
        "protocol": "udp"
      }
    ],
    "restart": "unless-stopped"
  }
]