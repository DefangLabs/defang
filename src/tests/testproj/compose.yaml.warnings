 ! port 4567: UDP ports default to 'host' mode (add 'mode: host' to silence)
 ! service "dfnx": secrets will be exposed as environment variables, not files (use 'environment' instead)
 ! unsupported compose extension: "x-unsupported"
 * Packaging the project files for dfnx at .