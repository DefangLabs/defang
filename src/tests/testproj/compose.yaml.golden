name: test
services:
  dfnx:
    build:
      context: ../../tests/testproj
      dockerfile: Dockerfile
      args:
        DNS: dfnx
      target: testproj
    deploy:
      resources:
        limits:
          cpus: 0.5
          memory: "536870912"
        reservations:
          cpus: 0.25
          memory: "268435456"
    environment:
      FOO: bar
    healthcheck:
      test:
        - CMD
        - curl
        - -f
        - http://localhost/
    networks:
      default: null
    ports:
      - mode: ingress
        target: 80
      - target: 1234
    restart: unless-stopped
    secrets:
      - source: dummy
        target: /run/secrets/dummy
networks:
  default:
    name: test_default
secrets:
  dummy:
    name: dummy
    external: true
x-unsupported: unsupported
